components/signoffs/signoffController.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const signoffDao = require('./signoffDao');
+const JSONStream = require('JSONStream');
+const {
+  handleErrors,
+  ValidationError,
+  ConflictingDataError,
+} = require('../../utils/controllerUtils');
+const { signoff } = require('../../utils/urlSlugs');
+const logger = require('../../utils/logger').logger;
+
+const getSignoff = async (req, res, next) => {
+  try {
+    const signoffs = await signoffDao.getByCaseIdentifier(
+      req.params.caseIdentifier
+    );
+    if (signoffs) {
+      res.status(200).json(signoffs);
+    } else {
+      res.status(404).end();
+    }
+  } catch (e) {
+    handleErrors(e, 'Error getting signoff(s) for case', logger, next);
+  }
+};
+
+const addSignoff = async (req, res, next) => {
+  try {
+    const existingSignoff = null;
+    //const existingSignoff = await signoffDao.getByCaseIdentifier(
```

Comment:
`signoffDao.getByCaseIdentifier` will return an array of 6+ signoffs (2 deliverableTypes * 3 signoffSteps * multiple users), so you might need to do some filtering here. It would be worth adding a couple of rows to V9000_test_data that have the same caseIdentifier as a previous row, but with different deliverableTypes and signoffSteps (and potentially multiple users and pass/fail values) during development to ensure you're catching all the edge cases we're going to encounter.